### DISCUSSION TITLE
[Feature] Global connection on network path

### DISCUSSION THREAD

**Discussion by Master-of-Mouse** [immich-app/immich#9146] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Searching for information I see many people running immich on a machine but saving the data on a NAS.

The official guide gives instructions on how to point to different paths on the same machine (https://immich.app/docs/guides/external-library).

I'm having a lot of difficulty configuring immich so that it runs alongside the database on my Debian VM with Docker but saves all the rest of the data to a Truenas share in another VM.

I am able to mount the share directly in Debian but best practices say to configure the docker-compose.yml file.

Would it be possible to update the official guide to describe how to do it even for those who are not familiar with Linux?
Or make a change to the structure of the docker-compose.yml and .env files so that, by entering the NAS IP and access credentials, immich uses that path to save all the data (cache, thumbnails, images, etc) .

Obviously it is clear that the shared folder on the NAS should not be touched except through immich otherwise everything could be ruined.

This type of configuration would also be very convenient for those who run immich on modest devices such as raspberry/zimaboard but have a NAS where they save the data of the various devices and then manage the backups from a single repository.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by mmomjian** [comment]
Thanks for your question. In order to do what you describe, you should mount the NAS on your system (using NFS, Samba, or SSHFS - you can Google a guide, but it sounds like you already have this set.) All you need to do is set the `UPLOAD_LOCATION` to that path. This would take a lot of work to implement in Immich for something that is already a basic part of the Linux kernel.
